src/dataflow/src/source/file.rs

```diff
@@ -91,6 +91,7 @@ impl SourceReader for FileSourceReader {
                         };
                         br.consume(chunk.len());
                         if chunk.len() > 0 {
+                            // `chunk` is always non-empty
```

Comment:
Does this invariant matter?

Reply:
yes, if we keep the `Data` non-empty as an invariant